staking/staking-specification.md

```diff
@@ -0,0 +1,815 @@
+# 0x protocol Staking specification
+
+## Table of Contents
+[1 Overview](#1-overview)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[1.1 Motivation](#11-motivation)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[1.2 Design Principles](#12-design-principles)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[1.3 Utility of Stake](#13-utility-of-stake)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[1.4 Staking Pools](#14-staking-pools)
+<br>[2 Architecture](#2-architecture)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[2.1 Read-Only Mode](#21-read-only-mode)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[2.2 Catastrophic Failure Mode](#22-catastrophic-failure-mode)
+<br>[3 Contract Migrations](#3-contract-migrations)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[3.1 Deploying the system](#31-deploying-the-system)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[3.2 Upgrading Staking Proxy](#32-upgrading-staking-proxy)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[3.3 Upgrading Staking Contract](#33-upgrading-staking-contract)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[3.4 Upgrading ZRX Vault](#34-upgrading-zrx-vault)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[3.5 Upgrading the WETH Asset Proxy](#35-upgrading-the-weth-asset-proxy)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[3.6 Handling upgrades to the ERC20 Proxy or ERC20 Asset Data](#36-handling-upgrades-to-the-erc20-proxy-or-erc20-asset-data)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[3.7 Setting Parameters](#37-setting-parameters)
+<br>[4 Epochs & Scheduling](#4-epochs--scheduling)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[4.1 Ending One Epoch, and Starting a New One](#41-ending-one-epoch-and-starting-a-new-one)
+<br>[5 Staking](#5-staking)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[5.1 Staking Pools](#51-staking-pools)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[5.2 Stake Status](#52-stake-status)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[5.3 Querying Stake](#53-querying-stake)
+<br>[6 Liquidity Incentives](#6-liquidity-incentives)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[6.1 Market Making](#61-market-making)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[6.2 Paying Liquidity Rewards (Finalization)](#62-paying-liquidity-rewards-finalization)
+<br>[7 Batch Calls](#7-batch-calls)
+<br>[8 Staking Events](#8-staking-events)
+<br>[9 Algorithms, Data Structures & Design Patterns](#9-algorithms-data-structures--design-patterns)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[9.1 Securing the Proxy Pattern](#91-securing-the-proxy-pattern)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[9.2 The Read-Only Proxy](#92-the-read-only-proxy)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[9.3 Tracking for Reward Balances for Pool Members](#93-tracking-for-reward-balances-for-pool-members)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;[9.4 Stake Management](#94-stake-management)
+
+## 1 Overview
+
+This spec outlines the design principles and constraints of the liquidity incentive mechanism, along with its architecture, implementation and usage.
+
+### 1.1 Motivation
+
+Staking aligns all market participants with the long-term mission and objectives of 0x. Specifically, it encourages user ownership among market makers by rewarding those who stake. Protocol fees redirect proceeds from arbitrage-driven gas auctions back to market makers.
```

Comment:
"Protocol fees redirect proceeds from arbitrage-driven gas auctions back to market makers."

This seems too specific to me. Some of the proceeds from arbitrage driven gas-auctions will be directed back, but so will the gas cost of normal takers